codegen/src/cli.ts

```diff
@@ -20,36 +21,23 @@ program
     defaultXataDirectory
   )
   .option('-o, --out <path>', 'A path to store your generated API client.', defaultOutputFile)
-  .action(async (xataDirectory, { out }) => {
-    const schema = join(xataDirectory, 'schema.json');
-    spinner.start('Checking schema...');
+  .option('-e, --env <path>', 'Path to the .env file to load.', '.env')
+  .action(async (xataDirectory, { out, env }) => {
+    dotenv.config({ path: env });
 
-    try {
-      await access(schema); // Make sure the schema file exists
-    } catch (e: any) {
-      if (e.code !== 'ENOENT') exitWithError(e);
-      spinner.info(
-        `You need to first install the Xata CLI and create a new database or pull the schema of an existing one. To learn more, visit ${chalk.blueBright(
-          'https://docs.xata.io/cli/getting-started'
-        )}.
-    `
-      );
-      exitWithError('No local Xata schema found.');
-      return;
-    }
+    spinner?.start('Checking schema...');
 
     try {
-      const dir = dirname(out);
-      await mkdir(dir, { recursive: true });
-      await generateWithOutput({ xataDirectory, outputFilePath: out, spinner });
-    } catch (e) {
-      exitWithError(e);
+      const databaseURL = getDatabaseURL();
+      const apiKey = getAPIKey();
+      if (databaseURL && apiKey) {
+        await generateFromAPI(out, { databaseURL, apiKey });
+      } else {
+        await generateFromLocalFiles(xataDirectory, out);
```

Comment:
Probably refactor to schemaFile (since we are removing the xataDirectory), still an offline copy of the online schema file could be used.